### PR TITLE
Add `PT016` documentation

### PR DESCRIPTION
**Pull request by harupy**
## Summary

- #2646
- Add `PT016` documentation

## Test Plan

N/A

### CODE REVIEW

File: crates/ruff/src/rules/flake8_pytest_style/rules/fail.rs [hunk 1]
```diff
@@ -8,6 +8,44 @@ use crate::checkers::ast::Checker;
 
 use super::helpers::{is_empty_or_null_string, is_pytest_fail};
 
+/// ## What it does
+/// Checks for ``pytest.fail`` calls without a message.
```
Comment (dhruvmanila) [inline crates/ruff/src/rules/flake8_pytest_style/rules/fail.rs:12]: nit:
```suggestion
/// Checks for `pytest.fail` calls without a message.
```

File: crates/ruff/src/rules/flake8_pytest_style/rules/fail.rs [hunk 2]
```diff
@@ -8,6 +8,44 @@ use crate::checkers::ast::Checker;
...
+/// def test_foo():
+///     pytest.fail("...")
+///
+///
+/// def test_bar():
+///     pytest.fail(reason="...")
+/// ```
+///
+/// ## References
+/// - [API Reference: pytest.fail](https://docs.pytest.org/en/latest/reference/reference.html#pytest-fail)
```
Comment (dhruvmanila) [inline crates/ruff/src/rules/flake8_pytest_style/rules/fail.rs:48]: nit:
```suggestion
/// - [API Reference: `pytest.fail`](https://docs.pytest.org/en/latest/reference/reference.html#pytest-fail)
```

**Review by dhruvmanila — APPROVED** [review]
Thanks!

**Comment by harupy** [comment]
@dhruvmanila Thanks for the suggestions!

**Pull request merged**